portal-api/src/utils/string.js

```diff
@@ -12,7 +14,48 @@ const hasValue = (str) => {
   return false;
 };
 
+/**
+ * Objective:
+  The objective of the isValidEmail function is to validate whether a
+  given email address is valid or not. It checks if the input is a non-empty
+  string and then uses the joi library to validate the email format.
+
+Inputs:
+  email: a string representing the email address to be validated
+
+Flow:
+  1. Check if the email is a non-empty string
+  2. Use the joi library to validate the email format
+  3. Return true if the email is valid, false otherwise
+
+Outputs:
+  true: if the email is valid
+  false: if the email is invalid
+
+Additional aspects:
+  The function uses the joi library for email validation
+  The email must be at least 3 characters long and in a valid email format
+  The function returns false if the input is not a non-empty string
+
+ * @param {String} email Email address as string
+ * @returns Boolean
+ */
+const isValidEmail = (email) => {
+  if (typeof email === 'string' && email.length > 0) {
+    const validation = joi
+      .string()
+      .email()
+      .required()
+      .validate(email);
+
```

Comment:
I think this could be simpler like @IgnatG suggested, in EXIP we just have this

```js

const schema = Joi.string().email();

const validation = schema.validate(email);

if (validation.error) {
  // return validation error
}
```

Reply:
Have simplified the compound condition.
Here we don't need to return any validation error just need to check if there is.